### ISSUE TITLE
James Cameron's AVATAR™: THE GAME [BLUS30374] - Foliage Flickering

### ISSUE THREAD

**Issue by Azonnali** [RPCS3/rpcs3#14202]
### Quick summary

The foliage flickers  on both my Windows PC and my Mac. Seems to be a revival of #11552 which should have been fixed by #11568. 

### Details

The foliage textures flicker a lot, everything else looks fine. 

As outlined in #11552 and #11568 I enabled "Allow GPU Host Labels" in the config yet this didnt resolve the problem.

I tried it on version 0.0.28-15355-1371bf89 on both of my machines (W11 and macOS): 

Windows 11 (22H2, build: 22621.1992)
Intel Arc A770 16GB (driver: 31.0.101.4575 [latest as of writing this])
Ryzen 5600
32 GB RAM

macOS 13.4.1 (c)
M1 Mac Mini
8 GB RAM

### Attach a log file

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12062298/RPCS3.log.gz)

### Attach capture files for visual issues

[BLUS30374_20230715193304_capture.zip](https://github.com/RPCS3/rpcs3/files/12062314/BLUS30374_20230715193304_capture.zip)

### System configuration

My Windows PC:
Windows 11 (22H2, build: 22621.1992)
Intel Arc A770 16GB (driver: 31.0.101.4575 [latest as of writing this])
Ryzen 5600
32 GB RAM

My Mac:
macOS 13.4.1 (c)
M1 Mac Mini
8 GB RAM

**Comment by kd-11** [comment]
That's because arc and M1 do not support the feature needed here. There's really no other way around it, other than using strict rendering mode and running at very low fps.

**Comment by kd-11** [comment]
> ·E 0:00:14.285845 RSX: INTEL driver does not support passthrough DMA buffers

Though now that I think about it, we're doing the check through a whitelist. Maybe arc fixed their issues? It's highly unlikely given that passthrough doesn't even work properly on NVIDIA cards. Only AMD has full support for this.

**Comment by Azonnali** [comment]
> That's because arc and M1 do not support the feature needed here. There's really no other way around it, other than using strict rendering mode and running at very low fps.

Just tested with strict rendering mode toggled on: Performance neither degraded nor did it fix the flickering. Happens on my Mac as well (I turned GPU Host Labels off again on both). I guess its because they do not support double precision floats? (attached new log file). Not sure if I should report this to Intel or not.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12062872/RPCS3.log.gz)

**Comment by kd-11** [comment]
> I guess its because they do not support double precision floats?
No, it's because the game actually starts overwriting VRAM based on when it sees labels.

You're right, even strict mode isn't strict enough here, because we'd actually get single digit framerates if we paused the pipe for real. The only solution is to configure the host GPU (arc/geforce/etc) to tell the game when it reaches the fence position. This means the GPU needs to have access to the vm memory. Most GPU drivers don't like our virtual memory because its meant for CPU use and comes in small memory mapped chunks.

Workarounds include having a separate CPU worker periodically unblock process threads but performance will be atrocious. I guess we can keep the ticket open until I have a working workaround committed at least.

**Comment by Guihmandela** [comment]
alguem sabe como eu faço para abrir o jogo ( Avatar James Cameron ) no RPCS3 ja tentei varias configuçoes e o jogo nao abre.

**Comment by kd-11** [comment]
I looked into this briefly a few weeks ago, M1 does expose the extension we need, but it doesn't seem to work right. Probably some other TBDR gotcha where I didn't flush the subpass when it wanted. I'll come up with a workaround to execute an event pump via the FIFO puller.

**Comment by ell1e** [comment]
I'm seeing the flickering as well in this game with Linux, RPCS3 0.0.32-1-1c16ada6 Alpha (Flathub), and [AMD Vega 6 graphics](https://www.notebookcheck.net/Ryzen-5-5560U-performance-debut-Beelink-SER5-mini-PC-review.762934.0.html). I enabled both "Allow GPU Host labels" and the strict mode. I'm guessing this is expected for now, but due to the GPU capabilities mentions above I just wanted to provide an additional data point in case for this GPU, this wasn't expected.

**Comment by digant73** [comment]
@ell1e, @Azonnali is that still an issue?

**Comment by Azonnali** [comment]
Hi there, sorry for the long no replies. 

I sold my Mac a while ago (and was only logged into github there), so I cannot comment on if the issue persists on Macs.

The issue does still occur on my A770 on Linux (Arch 6.19.10 and Mesa 26.0.4). ~~I can test it in Windows if you'd like as well.~~ 
Windows is broken as well (W11 25H2, Driver: 32.0.101.8724) 

Although I heavily suspect its a general Intel issue and they wont fix it - had the same thing happen with persistent issues on other emulators

**Comment by ell1e** [comment]
The hardware I saw this on was AMD. However, my machine is currently unavailable so I can't test again. But I should be able to retest in a few weeks.

**Comment by kd-11** [comment]
Host GPU labels regressed years ago so all devices are now affected by the bug again.

**Comment by Darkhost1999** [comment]
> Host GPU labels regressed years ago so all devices are now affected by the bug again.

I was not able to replicate the regression in the Far Cry series as foliage remains being fixed on my turing gpu. I don't own avatar to test that game.

**Comment by Jonathan44062** [comment]
idk what's referring to the broken foliage, works fine on my own

<img width="2559" height="1389" alt="Image" src="https://github.com/user-attachments/assets/b8286445-62b0-4b4e-8bee-0889dc76b4cc" />
even it worked in my potato GTX 1070 TI, then 2070, and now, the RTX 5070, only it's broken if Allow GPU Host Labels is not enable, the screenshot I shared is with the setting enabled I mentioned

**Comment by digant73** [comment]
@Azonnali, @ell1e please provide an update on this issue

**Comment by Azonnali** [comment]
@digant73 Still persists. Can provide videos if you'd like. Also let me know if I can help with logs or some other testing.

**Comment by digant73** [comment]
yes, provide new logs, rsx captures and video on current build if possible

**Comment by Azonnali** [comment]
Alright, latest version of RPCS3 (0.0.40-19376-61a26048 Alpha | master), running on the latest Intel drivers (32.0.101.8801)

Here the logs: [RPCS3.log.gz](https://github.com/user-attachments/files/27907873/RPCS3.log.gz)

Here the rsx-capture: [RSX-Capture](https://drive.google.com/file/d/1nLZUJ6AonFGuZRCv6JUnqtjca-KN7X_x/view?usp=drive_link) (I cant upload it to github for whatever reason, so here's a google drive link.

Here the video (as a gif but shows the issue):  <img width="426" height="240" alt="Image" src="https://github.com/user-attachments/assets/38072c2b-4dd3-4c05-af48-2848be194323" />
